Synthesis failed for asset-v1

Hello! Autosynth couldn't regenerate asset-v1. :broken_heart:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to a new branch 'autosynth-asset-v1'
Running synthtool
['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-03-29 03:37:05,458 synthtool > Executing /tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py.
2020-03-29 03:37:05,509 synthtool > Ensuring dependencies.
2020-03-29 03:37:05,513 synthtool > Cloning googleapis.
2020-03-29 03:37:06,275 synthtool > Pulling Docker image: gapic-generator-ruby:latest
latest: Pulling from gapic-images/gapic-generator-ruby
Digest: sha256:be51dd1d66d3afd09fe6cf572153e71c4d513ed0abfaaccc9c080faf29299cb8
Status: Image is up to date for gcr.io/gapic-images/gapic-generator-ruby:latest
2020-03-29 03:37:07,150 synthtool > Generating code for: google/cloud/asset/v1.
2020-03-29 03:37:08,207 synthtool > Failed executing docker run --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly --mount type=bind,source=/tmpfs/tmp/tmpcy2v6hkz/,destination=/out/ --rm --user 1000 --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly gcr.io/gapic-images/gapic-generator-ruby:latest --ruby-cloud-gem-name google-cloud-asset-v1 --ruby-cloud-title Cloud Asset V1 --ruby-cloud-description A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services. --ruby-cloud-env-prefix ASSET --ruby-cloud-grpc-service-config google/cloud/asset/v1/cloudasset_grpc_service_config.json:

google/cloud/orgpolicy/v1/orgpolicy.proto:22:1: warning: Import google/api/annotations.proto is unused.
google/cloud/orgpolicy/v1/orgpolicy.proto:20:1: warning: Import google/protobuf/empty.proto is unused.
google/identity/accesscontextmanager/v1/access_level.proto: File not found.
google/identity/accesscontextmanager/v1/access_policy.proto: File not found.
google/identity/accesscontextmanager/v1/service_perimeter.proto: File not found.
google/cloud/asset/v1/assets.proto:22:1: Import "google/identity/accesscontextmanager/v1/access_level.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:23:1: Import "google/identity/accesscontextmanager/v1/access_policy.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:24:1: Import "google/identity/accesscontextmanager/v1/service_perimeter.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:110:5: "google.identity.accesscontextmanager.v1.AccessPolicy" is not defined.
google/cloud/asset/v1/assets.proto:112:5: "google.identity.accesscontextmanager.v1.AccessLevel" is not defined.
google/cloud/asset/v1/assets.proto:114:5: "google.identity.accesscontextmanager.v1.ServicePerimeter" is not defined.
google/cloud/asset/v1/assets.proto:28:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/assets.proto:19:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/assets.proto:25:1: warning: Import google/protobuf/any.proto is unused.
google/cloud/asset/v1/asset_service.proto:23:1: Import "google/cloud/asset/v1/assets.proto" was not found or had errors.
google/cloud/asset/v1/asset_service.proto:195:3: "TimeWindow" is not defined.
google/cloud/asset/v1/asset_service.proto:201:12: "TemporalAsset" is not defined.
google/cloud/asset/v1/asset_service.proto:19:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/asset_service.proto:21:1: warning: Import google/api/field_behavior.proto is unused.
google/cloud/asset/v1/asset_service.proto:22:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/asset_service.proto:28:1: warning: Import google/type/expr.proto is unused.
google/cloud/asset/v1/asset_service.proto:20:1: warning: Import google/api/client.proto is unused.

2020-03-29 03:37:08,212 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py", line 37, in <module>
    "ruby-cloud-grpc-service-config": "google/cloud/asset/v1/cloudasset_grpc_service_config.json",
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 68, in ruby_library
    return self._generate_code(service, version, "ruby", **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 188, in _generate_code
    shell.run(docker_run_args)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly', '--mount', 'type=bind,source=/tmpfs/tmp/tmpcy2v6hkz/,destination=/out/', '--rm', '--user', '1000', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly', 'gcr.io/gapic-images/gapic-generator-ruby:latest', '--ruby-cloud-gem-name', 'google-cloud-asset-v1', '--ruby-cloud-title', 'Cloud Asset V1', '--ruby-cloud-description', 'A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services.', '--ruby-cloud-env-prefix', 'ASSET', '--ruby-cloud-grpc-service-config', 'google/cloud/asset/v1/cloudasset_grpc_service_config.json']' returned non-zero exit status 1.
Synthesis failed
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 482, in <module>
    main()
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 334, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 399, in _inner_main
    deprecated_execution=args.deprecated_execution,
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 278, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 369, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](https://sponge/282d4304-16e1-4e43-b437-a4797ddf5b70).

Autosynth is still having trouble generating asset-v1. :sob:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to a new branch 'autosynth-asset-v1'
Running synthtool
['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-03-30 03:37:24,388 synthtool > Executing /tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py.
2020-03-30 03:37:24,441 synthtool > Ensuring dependencies.
2020-03-30 03:37:24,446 synthtool > Cloning googleapis.
2020-03-30 03:37:24,877 synthtool > Pulling Docker image: gapic-generator-ruby:latest
latest: Pulling from gapic-images/gapic-generator-ruby
Digest: sha256:be51dd1d66d3afd09fe6cf572153e71c4d513ed0abfaaccc9c080faf29299cb8
Status: Image is up to date for gcr.io/gapic-images/gapic-generator-ruby:latest
2020-03-30 03:37:25,773 synthtool > Generating code for: google/cloud/asset/v1.
2020-03-30 03:37:26,853 synthtool > Failed executing docker run --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly --mount type=bind,source=/tmpfs/tmp/tmpgy70enmi/,destination=/out/ --rm --user 1000 --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly gcr.io/gapic-images/gapic-generator-ruby:latest --ruby-cloud-gem-name google-cloud-asset-v1 --ruby-cloud-title Cloud Asset V1 --ruby-cloud-description A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services. --ruby-cloud-env-prefix ASSET --ruby-cloud-grpc-service-config google/cloud/asset/v1/cloudasset_grpc_service_config.json:

google/cloud/orgpolicy/v1/orgpolicy.proto:22:1: warning: Import google/api/annotations.proto is unused.
google/cloud/orgpolicy/v1/orgpolicy.proto:20:1: warning: Import google/protobuf/empty.proto is unused.
google/identity/accesscontextmanager/v1/access_level.proto: File not found.
google/identity/accesscontextmanager/v1/access_policy.proto: File not found.
google/identity/accesscontextmanager/v1/service_perimeter.proto: File not found.
google/cloud/asset/v1/assets.proto:22:1: Import "google/identity/accesscontextmanager/v1/access_level.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:23:1: Import "google/identity/accesscontextmanager/v1/access_policy.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:24:1: Import "google/identity/accesscontextmanager/v1/service_perimeter.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:110:5: "google.identity.accesscontextmanager.v1.AccessPolicy" is not defined.
google/cloud/asset/v1/assets.proto:112:5: "google.identity.accesscontextmanager.v1.AccessLevel" is not defined.
google/cloud/asset/v1/assets.proto:114:5: "google.identity.accesscontextmanager.v1.ServicePerimeter" is not defined.
google/cloud/asset/v1/assets.proto:28:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/assets.proto:19:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/assets.proto:25:1: warning: Import google/protobuf/any.proto is unused.
google/cloud/asset/v1/asset_service.proto:23:1: Import "google/cloud/asset/v1/assets.proto" was not found or had errors.
google/cloud/asset/v1/asset_service.proto:195:3: "TimeWindow" is not defined.
google/cloud/asset/v1/asset_service.proto:201:12: "TemporalAsset" is not defined.
google/cloud/asset/v1/asset_service.proto:19:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/asset_service.proto:21:1: warning: Import google/api/field_behavior.proto is unused.
google/cloud/asset/v1/asset_service.proto:22:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/asset_service.proto:20:1: warning: Import google/api/client.proto is unused.
google/cloud/asset/v1/asset_service.proto:28:1: warning: Import google/type/expr.proto is unused.

2020-03-30 03:37:26,858 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py", line 37, in <module>
    "ruby-cloud-grpc-service-config": "google/cloud/asset/v1/cloudasset_grpc_service_config.json",
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 68, in ruby_library
    return self._generate_code(service, version, "ruby", **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 188, in _generate_code
    shell.run(docker_run_args)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly', '--mount', 'type=bind,source=/tmpfs/tmp/tmpgy70enmi/,destination=/out/', '--rm', '--user', '1000', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly', 'gcr.io/gapic-images/gapic-generator-ruby:latest', '--ruby-cloud-gem-name', 'google-cloud-asset-v1', '--ruby-cloud-title', 'Cloud Asset V1', '--ruby-cloud-description', 'A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services.', '--ruby-cloud-env-prefix', 'ASSET', '--ruby-cloud-grpc-service-config', 'google/cloud/asset/v1/cloudasset_grpc_service_config.json']' returned non-zero exit status 1.
Synthesis failed
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 482, in <module>
    main()
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 334, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 399, in _inner_main
    deprecated_execution=args.deprecated_execution,
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 278, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 369, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](https://sponge/c24ee4b0-170c-4453-80fa-e9292d5cb390).

Autosynth is still having trouble generating asset-v1. :sob:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to a new branch 'autosynth-asset-v1'
Running synthtool
['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-03-31 03:38:36,027 synthtool > Executing /tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py.
2020-03-31 03:38:36,078 synthtool > Ensuring dependencies.
2020-03-31 03:38:36,083 synthtool > Cloning googleapis.
2020-03-31 03:38:36,498 synthtool > Pulling Docker image: gapic-generator-ruby:latest
latest: Pulling from gapic-images/gapic-generator-ruby
Digest: sha256:be51dd1d66d3afd09fe6cf572153e71c4d513ed0abfaaccc9c080faf29299cb8
Status: Image is up to date for gcr.io/gapic-images/gapic-generator-ruby:latest
2020-03-31 03:38:37,367 synthtool > Generating code for: google/cloud/asset/v1.
2020-03-31 03:38:38,417 synthtool > Failed executing docker run --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly --mount type=bind,source=/tmpfs/tmp/tmpqc5_tlum/,destination=/out/ --rm --user 1000 --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly --mount type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly gcr.io/gapic-images/gapic-generator-ruby:latest --ruby-cloud-gem-name google-cloud-asset-v1 --ruby-cloud-title Cloud Asset V1 --ruby-cloud-description A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services. --ruby-cloud-env-prefix ASSET --ruby-cloud-grpc-service-config google/cloud/asset/v1/cloudasset_grpc_service_config.json:

google/cloud/orgpolicy/v1/orgpolicy.proto:22:1: warning: Import google/api/annotations.proto is unused.
google/cloud/orgpolicy/v1/orgpolicy.proto:20:1: warning: Import google/protobuf/empty.proto is unused.
google/identity/accesscontextmanager/v1/access_level.proto: File not found.
google/identity/accesscontextmanager/v1/access_policy.proto: File not found.
google/identity/accesscontextmanager/v1/service_perimeter.proto: File not found.
google/cloud/asset/v1/assets.proto:22:1: Import "google/identity/accesscontextmanager/v1/access_level.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:23:1: Import "google/identity/accesscontextmanager/v1/access_policy.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:24:1: Import "google/identity/accesscontextmanager/v1/service_perimeter.proto" was not found or had errors.
google/cloud/asset/v1/assets.proto:110:5: "google.identity.accesscontextmanager.v1.AccessPolicy" is not defined.
google/cloud/asset/v1/assets.proto:112:5: "google.identity.accesscontextmanager.v1.AccessLevel" is not defined.
google/cloud/asset/v1/assets.proto:114:5: "google.identity.accesscontextmanager.v1.ServicePerimeter" is not defined.
google/cloud/asset/v1/assets.proto:28:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/assets.proto:19:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/assets.proto:25:1: warning: Import google/protobuf/any.proto is unused.
google/cloud/asset/v1/asset_service.proto:23:1: Import "google/cloud/asset/v1/assets.proto" was not found or had errors.
google/cloud/asset/v1/asset_service.proto:195:3: "TimeWindow" is not defined.
google/cloud/asset/v1/asset_service.proto:201:12: "TemporalAsset" is not defined.
google/cloud/asset/v1/asset_service.proto:19:1: warning: Import google/api/annotations.proto is unused.
google/cloud/asset/v1/asset_service.proto:21:1: warning: Import google/api/field_behavior.proto is unused.
google/cloud/asset/v1/asset_service.proto:22:1: warning: Import google/api/resource.proto is unused.
google/cloud/asset/v1/asset_service.proto:28:1: warning: Import google/type/expr.proto is unused.
google/cloud/asset/v1/asset_service.proto:20:1: warning: Import google/api/client.proto is unused.

2020-03-31 03:38:38,422 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/tmpfs/src/git/autosynth/working_repo/google-cloud-asset-v1/synth.py", line 37, in <module>
    "ruby-cloud-grpc-service-config": "google/cloud/asset/v1/cloudasset_grpc_service_config.json",
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 68, in ruby_library
    return self._generate_code(service, version, "ruby", **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/gcp/gapic_microgenerator.py", line 188, in _generate_code
    shell.run(docker_run_args)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['docker', 'run', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/asset/v1/,destination=/in/google/cloud/asset/v1/,readonly', '--mount', 'type=bind,source=/tmpfs/tmp/tmpqc5_tlum/,destination=/out/', '--rm', '--user', '1000', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/common_resources.proto,destination=/in/google/cloud/common_resources.proto,readonly', '--mount', 'type=bind,source=/home/kbuilder/.cache/synthtool/googleapis/google/cloud/orgpolicy/v1/orgpolicy.proto,destination=/in/google/cloud/orgpolicy/v1/orgpolicy.proto,readonly', 'gcr.io/gapic-images/gapic-generator-ruby:latest', '--ruby-cloud-gem-name', 'google-cloud-asset-v1', '--ruby-cloud-title', 'Cloud Asset V1', '--ruby-cloud-description', 'A metadata inventory service that allows you to view, monitor, and analyze all your GCP and Anthos assets across projects and services.', '--ruby-cloud-env-prefix', 'ASSET', '--ruby-cloud-grpc-service-config', 'google/cloud/asset/v1/cloudasset_grpc_service_config.json']' returned non-zero exit status 1.
Synthesis failed
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 482, in <module>
    main()
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 334, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 399, in _inner_main
    deprecated_execution=args.deprecated_execution,
  File "/tmpfs/src/git/autosynth/autosynth/synth.py", line 278, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 369, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](https://sponge/6e5fe7bf-91e4-410c-aa7f-92fbb03fdbea).

--- a/google-cloud-asset-v1/synth.py
+++ b/google-cloud-asset-v1/synth.py
@@ -28,6 +28,10 @@ library = gapic.ruby_library(
     extra_proto_files=[
         "google/cloud/common_resources.proto",
         "google/cloud/orgpolicy/v1/orgpolicy.proto",
+        "google/identity/accesscontextmanager/type/device_resources.proto",
+        "google/identity/accesscontextmanager/v1/access_level.proto",
+        "google/identity/accesscontextmanager/v1/access_policy.proto",
+        "google/identity/accesscontextmanager/v1/service_perimeter.proto",
     ],
     generator_args={
         "ruby-cloud-gem-name": "google-cloud-asset-v1",

--- a/google-cloud-asset/synth.py
+++ b/google-cloud-asset/synth.py
@@ -24,6 +24,14 @@ logging.basicConfig(level=logging.DEBUG)
 gapic = gcp.GAPICMicrogenerator()
 library = gapic.ruby_library(
     "asset", "v1",
+    extra_proto_files=[
+        "google/cloud/common_resources.proto",
+        "google/cloud/orgpolicy/v1/orgpolicy.proto",
+        "google/identity/accesscontextmanager/type/device_resources.proto",
+        "google/identity/accesscontextmanager/v1/access_level.proto",
+        "google/identity/accesscontextmanager/v1/access_policy.proto",
+        "google/identity/accesscontextmanager/v1/service_perimeter.proto",
+    ],
     generator_args={
         "ruby-cloud-gem-name": "google-cloud-asset",
         "ruby-cloud-title": "Cloud Asset",

